tests/components/application-connector/test/compass-runtime-agent/testkit/applications/secretcomparator_test.go

```diff
@@ -0,0 +1,116 @@
+package applications
+
+import (
+	"context"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"testing"
+)
+
+func TestCompare(t *testing.T) {
+
+	t.Run("should return true if secrets are equal", func(t *testing.T) {
+		//given
+		coreV1 := fake.NewSimpleClientset()
+		secretComparator, err := NewSecretComparator(require.New(t), coreV1, "test", "kyma-integration")
+		require.NoError(t, err)
+		createFakeCredentialsSecret(t, coreV1.CoreV1().Secrets("test"), "expected", "test")
+		createFakeCredentialsSecret(t, coreV1.CoreV1().Secrets("kyma-integration"), "actual", "kyma-integration")
+
+		//when
+		err = secretComparator.Compare("expected", "actual")
+
+		// then
+		require.NoError(t, err)
+	})
+
```

Comment:
Notice, that `require.Assertions` is used. When values compared with `Equal` are different `t.FailNow` is called and tests execution is interrupted.  So that if we added such a test it would always fail.

Please mind the following:
- In the initial approach the comparer interface was returning bool. The implementation was simply comparing two objects and returning true or false. The problem occurs when you need to print the comparison result in the actual test. The caller gets false and needs to fail the test and print some information to the log file that allows developer to find why the comparison failed. But what should be printed to the log file?  Ok, we can write compared objects. But let's think for a while. What would be the user experience? Maybe I'm missing something, but passing `require.Assertions` to the Comparer solves this problem.   
- This code is a tool used for implementing actual test cases. 
- There is kind of "what will test the test" problem. This code is complex enough to be tested, but we cannot cover all cases by design. I'm not in favour of "all or nothing" approach and I think it gives enough value.